### DISCUSSION TITLE
How to use with Typescript and/or use className

### DISCUSSION THREAD

**Discussion by robhaven** [ariakit/ariakit#759] [Questions] [answered]
Reading the docs I would think I could use `className` for the `<Button>` component.

When I try to add className attribute:
![image](https://user-images.githubusercontent.com/71856696/95638253-cc831d80-0a48-11eb-95c6-7616a7330e71.png)

I did a bit of digging in `reakit/ts/Button/Button.d.ts` and the imports but didn't see where `className` prop is. It seems to be mentioned several times in the docs so perhaps I'm missing something?

My setup is basically create-react-app with typescript and then reakit added:

```package.json
{
  "name": "rapidapi-react-typescript",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "@types/jest": "^24.0.0",
    "@types/node": "^12.0.0",
    "@types/react": "^16.9.0",
    "@types/react-dom": "^16.9.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "3.4.3",
    "reakit": "^1.2.5",
    "typescript": "~3.7.2"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

**Comment by diegohaz** [comment, marked as answer]
It should work!

Silly question, but just to make sure: have you tried passing a string to `className`? Your screenshot is showing a boolean value, which is not valid.

**Reply by robhaven** [reply]
You are absolutely correct -- I just gave up too soon lol. I suck at TS :( Thanks @diegohaz -- I'm pretty amazed at how well a11y is implemented here. I've done this by hand for a long time and the idea of having a set of components that have this baked in for free, yet flexible like this is just amazing!


Ok, unrelated, but I also noticed that you can certainly use CSS Modules fine yet the docs don't brag on that:

```
<Button className={styles.Button}>{label}</Button>
```

I was looking at the styling docs and was pretty sure you could, but noticed it's not mentioned:
https://reakit.io/docs/styling/#inline-styles

I see the md file -- should I create a PR for that?

![image](https://user-images.githubusercontent.com/71856696/95641277-37872100-0a56-11eb-88b7-3903c3f33d32.png)


I've used CSS Modules at 3 jobs in a row, so religious wars or opinions aside, I think pointing out it's totally possible would help adoption no?

Btw, when I'm not working I'm rolling my own framework AgnosticUI: https://github.com/AgnosticUI/agnosticui and I'm very seriously considering replacing or adding the React package with a reakit version since the two libraries have absolutely no venn overlap otherwise. Very excited about what you've got here sir!

**Reply by diegohaz** [reply]
@robhaven Nice! Feel free to send a PR with the CSS modules addition.

You will want to do ` ```jsx static` instead of ` ```jsx` since `import styles from "./Example.module.css";` won't compile in the website.
